### ISSUE TITLE
[BUG]"No Such File or Directory"

### ISSUE THREAD

**Issue by yaokunxu** [wrh14/Learning_to_Invert#1]
You work in the paper is defnitely awesome.i want to reproduce your outcome but encounter with some problems.I am wondering whether youc could help me to fix this.Any information from you will be gratitude.

+Description+
I run the command from Readme.md and has seaarch for related problem.
As the Error indicated,some unexcepted is containing in the ./fl_settings/cola*.pt
+------------+

+environment+
Ubuntu 20.04
python 3.9.19
torch 2.2.2+cuda
+---------------+

+command+
python main_learn_dlg_large_model.py --epochs 100 --batch_size 16 --dataset cola-hash --shared_model BERT --model NLPMLP-600-1000 --lr 1e-3 --leak_mode None
+-----------+

+output+
[24-04-30 10:50:46 root] Namespace(dataset='cola-hash', model='NLPMLP-600-1000', shared_model='BERT', lr=0.001, epochs=100, batch_size=16, seed=0, leak_mode='None', get_validation_rec_first_thou=None, resume=None)
[24-04-30 10:50:46 root] logs are saved at print_logs/cola-hash_BERT_NLPMLP-600-1000_None_0.001_100_16_0.txt
[24-04-30 10:50:47 root] model size: 86042882
[24-04-30 10:50:47 root] loading dataset...
[24-04-30 10:50:48 datasets] PyTorch version 2.2.2 available.
Traceback (most recent call last):
  File "/data/xhd/Learning_to_Invert/main_learn_dlg_large_model.py", line 455, in <module>
    dst_train = torch.load(f"./fl_settings/cola_train_dataset.pt")
  File "/data/xhd/.conda/envs/lamp/lib/python3.9/site-packages/torch/serialization.py", line 1026, in load
    return _load(opened_zipfile,
  File "/data/xhd/.conda/envs/lamp/lib/python3.9/site-packages/torch/serialization.py", line 1438, in _load
    result = unpickler.load()
  File "/data/xhd/.conda/envs/lamp/lib/python3.9/site-packages/datasets/table.py", line 1028, in __setstate__
    table = _memory_mapped_arrow_table_from_file(path)
  File "/data/xhd/.conda/envs/lamp/lib/python3.9/site-packages/datasets/table.py", line 64, in _memory_mapped_arrow_table_from_file
    opened_stream = _memory_mapped_record_batch_reader_from_file(filename)
  File "/data/xhd/.conda/envs/lamp/lib/python3.9/site-packages/datasets/table.py", line 49, in _memory_mapped_record_batch_reader_from_file
    memory_mapped_stream = pa.memory_map(filename)
  File "pyarrow/io.pxi", line 1053, in pyarrow.lib.memory_map
  File "pyarrow/io.pxi", line 1000, in pyarrow.lib.MemoryMappedFile._open
  File "pyarrow/error.pxi", line 154, in pyarrow.lib.pyarrow_internal_check_status
  File "pyarrow/error.pxi", line 91, in pyarrow.lib.check_status
FileNotFoundError: [Errno 2] Failed to open local file '/home/rw565/data/glue/cola/1.0.0/dacbe3125aa31d7f70367a07a8a9e72a5a0bfeb5fc42e75c9db75b96da6053ad/cache-6a7599bc734506da.arrow'. Detail: [errno 2] No such file or directory
+------+

**Comment by baimuchu** [comment]
yes, I also have the problem. It seems that pre-train model (.pt file) direct a author's local address.

**Comment by wrh14** [comment]
Hi @yaokunxu and @baimuchu, thanks for bring this issue up to me. 

I reproduced the error you were meeting and yes it is because of some missing local files. Unfortunately, I permanently lost those files either. Alternatively, I am working on to generate these checkpoints in fl_settings from scratch and will keep updating you once I am done.

**Comment by yaokunxu** [comment]
> Hi @yaokunxu and @baimuchu, thanks for bring this issue up to me.
> 
> I reproduced the error you were meeting and yes it is because of some missing local files. Unfortunately, I permanently lost those files either. Alternatively, I am working on to generate these checkpoints in fl_settings from scratch and will keep updating you once I am done.

Thanks a lot. With great graditude

**Comment by wrh14** [comment]
Hi @yaokunxu and @baimuchu , sorry for this delay; it took some time to reconstruct those intermediate files and check the results after. The changes can be summarized below:
- I have updated the folder `fl_settings` together with some necessary edits in code
- [README.md](https://github.com/wrh14/Learning_to_Invert/blob/main/README.md) has been updated with more details for the set-up.

Let me know if you have any further question!
